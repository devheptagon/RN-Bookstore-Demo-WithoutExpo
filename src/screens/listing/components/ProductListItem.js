import React from "react";
import PropTypes from "prop-types";
import { StyleSheet } from "react-native";
import ProductInfo from "./ProductInfo";
import FlexView from "../../components/FlexView";
import Theme from "../../../constants/Theme";
import { FlexDirections } from "../../../constants/Enums";

export class ProductListItem extends React.PureComponent {
  render() {
    const { dataItem } = this.props;
    return (
      <FlexView flexDirection={FlexDirections.Row} style={styles.Wrapper}>
        <ProductInfo dataItem={dataItem} />
      </FlexView>
    );
  }
}

const styles = StyleSheet.create({
  Wrapper: {
    height: 120,
    borderTopWidth: Theme.General.Border.Width,
    borderBottomWidth: Theme.General.Border.Width,
    borderTopColor: Theme.ProductList.Item.BorderColor,
    borderBottomColor: Theme.ProductList.Item.BorderColor,
    backgroundColor: Theme.ProductList.Item.BackgroundColor
  }
});

ProductListItem.propTypes = {
  dataItem: PropTypes.object
};

export default ProductListItem;
