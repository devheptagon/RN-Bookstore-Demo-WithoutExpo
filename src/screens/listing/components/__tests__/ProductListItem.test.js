import React from "react";
import { ProductListItem } from "../ProductListItem";
import renderer from "react-test-renderer";

test("ProductListItem renders correctly", () => {
  const dataItem = {
    id: "GqTwl7-WZtUC",
    volumeInfo: {
      imageLinks: {
        smallThumbnail:
          "http://books.google.com/books/content?id=GqTwl7-WZtUC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
        thumbnail:
          "http://books.google.com/books/content?id=GqTwl7-WZtUC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
      }
    }
  };

  const tree = renderer
    .create(<ProductListItem dataItem={dataItem} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
