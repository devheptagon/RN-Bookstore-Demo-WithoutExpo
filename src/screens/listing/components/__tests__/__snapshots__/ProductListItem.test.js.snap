// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListItem renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderBottomColor": "#f5f5f5",
      "borderBottomWidth": 1,
      "borderTopColor": "#f5f5f5",
      "borderTopWidth": 1,
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "padding": 5,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "http://books.google.com/books/content?id=GqTwl7-WZtUC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
            }
          }
          style={
            Array [
              Object {
                "borderColor": "#eee",
                "borderWidth": 2,
              },
              Object {
                "height": "100%",
                "width": "100%",
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 3,
          "flexDirection": "column",
          "padding": 10,
        }
      }
    />
  </View>
</View>
`;
